packages/libsolv/libsolv.spec

```diff
@@ -1,60 +1,59 @@
-%global python3_pkgversion 3.11
-%global __python3 /usr/bin/python3.11
-
-# explicitly define, as we build on top of an scl, not inside with scl_package
-%if 0%{?scl:1}
-%global scl_prefix %{scl}-
-%global python3_sitearch /opt/theforeman/tfm-pulpcore/root/usr/lib64/python3.8/site-packages/
-%global python3_version %python38python3_version
-%global __os_install_post %python38_os_install_post
-%global __python_requires %python38_python_requires
-%global __python_provides %python38_python_provides
-%global __python3 %python38__python
-%endif
-
-# Our EL8 buildroots default to Python 3.8, but let's also build 3.6, just to be safe
-# to make dnf happy
-%if 0%{?rhel} == 8
-%bcond_without python36
-%else
-%bcond_with python36
-%endif
-
 %global libname solv
 
-# Only build Python2 bindings on EL7
-%if 0%{?rhel} == 7
-%bcond_without python2_bindings
-%else
-%bcond_with python2_bindings
-%endif
-
-%bcond_without python3_bindings
-%bcond_with    perl_bindings
+%bcond_without python_bindings
+%bcond_without perl_bindings
 %bcond_without ruby_bindings
 # Creates special prefixed pseudo-packages from appdata metadata
 %bcond_without appdata
 # Creates special prefixed "group:", "category:" pseudo-packages
 %bcond_without comps
+%bcond_without conda
 # For rich dependencies
 %bcond_without complex_deps
+%if 0%{?rhel}
+%bcond_with helix_repo
+%bcond_with suse_repo
+%bcond_with debian_repo
+%bcond_with arch_repo
+# For handling deb + rpm at the same time
+%bcond_with multi_semantics
+%bcond_with zchunk
+%else
 %bcond_without helix_repo
 %bcond_without suse_repo
 %bcond_without debian_repo
 %bcond_without arch_repo
 # For handling deb + rpm at the same time
 %bcond_without multi_semantics
-%bcond_with    zchunk
-%bcond_with    zstd
+%bcond_without zchunk
+%endif
+%bcond_without zstd
+
+#global commitnum 2901
+#global commit 47fbaa2a0892866d30ec0e1b4c885532d0aca7b8
+#global shortcommit %%(c=%%{commit}; echo ${c:0:7})
+
+%define __cmake_switch(b:) %{expand:%%{?with_%{-b*}:ON}}%{expand:%%{!?with_%{-b*}:OFF}}
 
 Name:           lib%{libname}
-Version:        0.7.22
+Version:        0.7.20
```

Comment:
Why are we downgrading things?

Reply:
additionally: how are we going to handle EL9 which ships libsolv-0.7.24 today.

Reply:
and what about https://github.com/pulp/pulp_rpm/blob/main/requirements.txt#L7

Reply:
I have confirmed with pulp that while the requirements.txt for pip is 0.7.21, the functionality required should be back-ported into the OS provided libsolv.  We should be able to modify that requirements.txt via spec to loosen the requirement. (it's not exactly true in a purely upstream pip installed context, hence the need for spec modificaiton).

Ideally, we should probably _stop_ shipping libsolv on el9.  Revert back to using system libsolv, and only ship 3.11 bindings for it.  As far as "how", i'm not entirely sure.  I need to do a little research in this space.

Reply:
Ok, so after some research and some help from fedora-devel matrix channel, we can build the python binding in a statically linked way.  I played around with this locally, and it also appears to be the way solv is distributed by pypi.  Therefore it won't override system libsolv, but won't use it either.  Meaning we can safely decouple from system libsolv, unless we have good reason to keep it coupled.

Reply:
Yup :D best way out of this mess I can figure out.